packages/flutter_tools/lib/src/commands/run.dart

```diff
@@ -284,6 +284,10 @@ class RunCommand extends RunCommandBase {
     if (!runningWithPrebuiltApplication) {
       await super.validateCommand();
     }
+    devices = await findAllTargetDevices();
+    if (devices == null) {
+      throwToolExit(null);
```

Comment:
Should this give a message like "Failed to find a target device" ?

Reply:
In `findAllTargetDevices()`, before returning `null`, the appropriate `userMessages` will be printed to stdout; in this case:

```sh
~/git/flutter/examples/hello_world$ flutter run
Building flutter tool...
No supported devices connected.
```

So we're covered. I'll add a test verifying this (we have one for attach, but not for run) though.

Reply:
Added a test, PTAL